ksql-parser/src/main/java/io/confluent/ksql/schema/ksql/DefaultSqlValueCoercer.java

```diff
@@ -35,46 +35,57 @@
           .put(SqlBaseType.DOUBLE, Number::doubleValue)
           .build();
 
-  public <T> Optional<T> coerce(final Object value, final Schema targetSchema) {
+  public <T> Optional<T> coerce(final Object value, final SqlType targetType) {
+    if (targetType.baseType() == SqlBaseType.ARRAY
```

Comment:
There were existing test cases that tested this call failed for these three types, but the actual exception was thrown elsewhere by one of the `SchemaConverter` impls, which no longer throws. Hence adding this check.